src/components/sql-editor/SQLEditor.tsx

```diff
@@ -0,0 +1,284 @@
+import { CodeEditor, Monaco, monacoTypes } from '@grafana/ui';
+import React, { useMemo } from 'react';
+import { getStatementPosition } from './standardSql/getStatementPosition';
+import { getStandardSuggestions } from './standardSql/getStandardSuggestions';
+import { initSuggestionsKindRegistry, SuggestionKindRegistyItem } from './standardSql/suggestionsKindRegistry';
+import { CompletionItemPriority, CustomSuggestion, PositionContext, SQLCompletionItemProvider } from './types';
+import { getSuggestionKinds } from './utils/getSuggestionKind';
+import { linkedTokenBuilder } from './utils/linkedTokenBuilder';
+import { StatementPosition, SuggestionKind } from './utils/types';
+import { getTableToken } from './utils/tokenUtils';
+import { TRIGGER_SUGGEST } from './utils/misc';
+import { LinkedToken } from './utils/LinkedToken';
+import { v4 } from 'uuid';
+import { Registry } from '@grafana/data';
+import {
+  FunctionsRegistryItem,
+  OperatorsRegistryItem,
+  StatementPositionResolversRegistryItem,
+  SuggestionsRegistyItem,
+} from './standardSql/types';
+import {
+  initFunctionsRegistry,
+  initOperatorsRegistry,
+  initStandardSuggestions,
+} from './standardSql/standardSuggestionsRegistry';
+import { initStatementPositionResolvers } from './standardSql/statementPositionResolversRegistry';
+
+const STANDARD_SQL_LANGUAGE = 'sql';
+
+interface LanguageDefinition extends monacoTypes.languages.ILanguageExtensionPoint {
+  // TODO: Will allow providing a custom language definition.
+  loadLanguage?: (module: any) => Promise<void>;
+  // Provides API for customizing the autocomplete
+  completionProvider?: (m: Monaco) => SQLCompletionItemProvider;
+}
+
+interface SQLEditorProps {
+  query: string;
+  onChange: (q: string) => void;
+  language?: LanguageDefinition;
+}
+
+const defaultTableNameParser = (t: LinkedToken) => t.value;
+
+interface LanguageRegistries {
+  functions: Registry<FunctionsRegistryItem>;
+  operators: Registry<OperatorsRegistryItem>;
+  suggestionKinds: Registry<SuggestionKindRegistyItem>;
+  positionResolvers: Registry<StatementPositionResolversRegistryItem>;
+}
+
+const LANGUAGES_CACHE = new Map<string, LanguageRegistries>();
+const INSTANCE_CACHE = new Map<string, Registry<SuggestionsRegistyItem>>();
```

Comment:
for these two caches (`LANGUAGES_CACHE` and `INSTANCE_CACHE`).. do you plan to remove items from them at some point in some way, or they will just grow "forever"?

Reply:
Yes, I plan to clean those caches

Reply:
https://github.com/grafana/grafana-experimental/pull/9